core/src/test/java/com/sequenceiq/cloudbreak/controller/StackCreatorServiceTest.java

```diff
@@ -399,6 +425,113 @@ public void testFillInstanceMetadataWhenManager() {
+        stack.setType(StackType.DATALAKE);
+        InstanceGroupNetwork network = new InstanceGroupNetwork();
+        network.setCloudPlatform("aws");
+        network.setAttributes(Json.silent(Map.of(SUBNET_IDS, List.of("SUB_1", "SUB_2", "SUB_3"))));
+        InstanceGroup masterGroup = getARequestGroup("master", 2, InstanceGroupType.CORE);
+        masterGroup.setInstanceGroupNetwork(network);
+        InstanceGroup gatewayGroup = getARequestGroup("gateway", 2, InstanceGroupType.GATEWAY);
+        gatewayGroup.setInstanceGroupNetwork(network);
+        InstanceGroup coreGroup = getARequestGroup("core", 3, InstanceGroupType.CORE);
+        coreGroup.setInstanceGroupNetwork(network);
+        InstanceGroup auxiliaryGroup = getARequestGroup("auxiliary", 1, InstanceGroupType.CORE);
+        auxiliaryGroup.setInstanceGroupNetwork(network);
+        InstanceGroup idbrokerGroup = getARequestGroup("idbroker", 2, InstanceGroupType.CORE);
+        idbrokerGroup.setInstanceGroupNetwork(network);
+        stack.setInstanceGroups(Set.of(masterGroup, coreGroup, auxiliaryGroup, idbrokerGroup, gatewayGroup));
+        doReturn(subnetAzPairs(3)).when(multiAzCalculatorService).prepareSubnetAzMap(environmentResponse);
+
+
+        underTestSpy.fillInstanceMetadata(environmentResponse, stack);
+        verify(multiAzCalculatorService, times(4)).calculateByRoundRobin(anyMap(), any(InstanceGroupNetwork.class), anySet());
+        verify(underTestSpy, times(1)).datalakeMultiAZCalculator(any(), any(), any(), any());
+        reset(multiAzCalculatorService);
+
+        stack.setInstanceGroups(Set.of(auxiliaryGroup, gatewayGroup));
+        underTest.fillInstanceMetadata(environmentResponse, stack);
+        ArgumentCaptor<Set<InstanceMetaData>> capture = ArgumentCaptor.forClass(Set.class);
+        verify(multiAzCalculatorService, times(1)).calculateByRoundRobin(anyMap(), any(InstanceGroupNetwork.class), capture.capture());
+        Assert.assertEquals(3, capture.getValue().size());
+        capture.getValue().stream().allMatch(instanceMetaData -> instanceMetaData.getId() == 1L);
+    }
+
+    @Test
+    void testFillInstanceMetadataForDH() {
+        Stack stack = new Stack();
+        stack.setType(StackType.WORKLOAD);
+        InstanceGroupNetwork network = new InstanceGroupNetwork();
+        network.setCloudPlatform("aws");
+        network.setAttributes(Json.silent(Map.of(SUBNET_IDS, List.of("SUB_1", "SUB_2", "SUB_3"))));
+        InstanceGroup masterGroup = getARequestGroup("master", 2, InstanceGroupType.CORE);
+        masterGroup.setInstanceGroupNetwork(network);
+        InstanceGroup gatewayGroup = getARequestGroup("gateway", 2, InstanceGroupType.GATEWAY);
+        gatewayGroup.setInstanceGroupNetwork(network);
+        InstanceGroup coreGroup = getARequestGroup("core", 3, InstanceGroupType.CORE);
+        coreGroup.setInstanceGroupNetwork(network);
+        InstanceGroup auxiliaryGroup = getARequestGroup("auxiliary", 1, InstanceGroupType.CORE);
```

Comment:
Can we have a test case without `auxiliary` group?

Reply:
added.